### PR TITLE
Init Treedome Package

### PR DESCRIPTION
**Pull request by tengkuizdihar**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nagy — CHANGES_REQUESTED** [review]
Thanks, can you follow the guidelines for the commit message at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions ?

File: pkgs/applications/misc/treedome/default.nix [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  buildPhase = ''
+    runHook preBuild
+
+    cargo tauri build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
```
Comment (nagy) [inline pkgs/applications/misc/treedome/default.nix:97]: This needs `runHook preInstall` and `runHook postInstall`.

File: pkgs/applications/misc/treedome/default.nix [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    cp target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/bin/treedome $out/bin/treedome
+    cp -R target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/share/** $out/share/
+  '';
+
+  meta = with lib; {
+    description = "A local-first, encrypted, note taking application with tree-like structures, all written and saved in markdown.";
+    homepage = "https://gitlab.com/treedome/treedome";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ tengkuizdihar ];
```
Comment (nagy) [inline pkgs/applications/misc/treedome/default.nix:109]: Can you add `mainProgram = "treedome";` ?

**Comment by tengkuizdihar** [comment]
Will do, sorry this shouldve been a draft pr lol, its not ready yet im
still tinkering with it

File: pkgs/applications/misc/treedome/default.nix [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    mkdir -p $out/share/
+
+    cp target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/bin/treedome $out/bin/treedome
+    cp -R target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/share/** $out/share/
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A local-first, encrypted, note taking application with tree-like structures, all written and saved in markdown.";
```
Comment (kilianar) [inline pkgs/applications/misc/treedome/default.nix:108]: The description should not include a period at the end. See https://nixos.org/manual/nixpkgs/stable/#var-meta-description

**Comment by tengkuizdihar** [comment]
oh no

**Pull request closed without merging**